src/api/model/modelApi.ts

```diff
@@ -14,57 +14,91 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
+import _ from 'lodash';
 
-export class ModelApi extends TransactionApi {
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
   async installModels(database: string, engine: string, models: Model[]) {
-    const action: InstallAction = {
-      type: 'InstallAction',
-      sources: models,
-    };
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """ ${model.value} """`;
+    });
 
-    return await this.runActions(database, engine, [action], false);
+    return await this.exec(database, engine, queries.join('\n'), [], false);
+  }
+
+  async installModelsAsync(database: string, engine: string, models: Model[]) {
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """ ${model.value} """`;
+    });
+
+    return this.execAsync(database, engine, queries.join('\n'), [], false);
   }
 
   async listModels(database: string, engine: string) {
-    const action: ListSourceAction = {
-      type: 'ListSourceAction',
-    };
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const outName = `models${randInt}`;
+    const rsp = await this.exec(
+      database,
+      engine,
+      `def output:${outName}[name] = rel:catalog:model(name, _)`,
+    );
+
+    const models = rsp.results.map(result => {
+      if (result.relationId.includes('/:output/:models')) {
+        return result.table.toArray().map(col => {
+          return col.v1;
```

Comment:
I guess no reason to rely on the column name here.

Reply:
table is an array of `[{"v1": "model name"}, {"v1": ...}]` so I think we need `array[i].v1` to get the model name unless I'm missing something

Reply:
There's no point in hardcoding on the name of the column `v1`. You can just get it by the index. `.toArray()[0]`